pandas/plotting/_matplotlib/core.py

```diff
@@ -124,13 +126,31 @@ def __init__(
         table=False,
         layout=None,
         include_bool=False,
+        column: IndexLabel | None = None,
         **kwds,
     ):
 
         import matplotlib.pyplot as plt
 
         self.data = data
-        self.by = by
+        self.by = com.maybe_make_list(by)
+
+        # For `hist` plot, need to get grouped original data before `self.data` is
+        # updated later
+        if self.by and self._kind == "hist":
+            self._grouped = data.groupby(self.by)
+
+        # Assign the rest of columns into self.columns if by is explicitly defined
+        # while column is not, so as to keep the same behaviour with current df.hist
+        # or df.boxplot.
+        if self.by and column is None:
```

Comment:
Minor thing, but since you've got two conditions with `self.by`, maybe a `if self.by:` that contains these two `if`. Not sure, up to you, but feels like it could group the code executed when `by` is present in a more readable way.

Reply:
Sure! And I also did small changes and realised that we could put `column` check part separately, because no matter if `by` input is None or not, `column` should be updated accordingly.